src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -264,25 +272,20 @@ private int WriteTags(ref byte[] bytes, int offset)
             }
             else
             {
-                count = 0;
                 offset += MessagePackBinary.WriteMapHeader(ref bytes, offset, count);
             }
 
-            foreach (var property in GetAdditionalTags())
+            foreach (var property in additionalTags)
             {
                 var value = property.Getter(this);
```

Comment:
only meh for me is calling the property getter twice for each additionalTags

Reply:
I know 😞 but I can think of only 3 possibilities:
- Call the property getter twice
- Store the values in an array (adds a heap allocation on netfx/netstandard, could be allocated on the stack on netcoreapp3.1)
- Serialize empty tags/metrics (not actually sure it's possible for metrics since it's a numeric type)

Reply:
mmm yeah, you're right, I guess the best is to leave it this way.